backend/src/server/routes/v3/signup-router.ts

```diff
@@ -23,8 +25,20 @@ export const registerSignupRouter = async (server: FastifyZodProvider) => {
       }
     },
     handler: async (req) => {
-      await server.services.signup.beginEmailSignupProcess(req.body.email);
-      return { message: `Sent an email verification code to ${req.body.email}` };
+      const { email } = req.body;
+      const serverCfg = await getServerCfg();
+
+      if (serverCfg?.allowSpecificDomainSignUp) {
```

Comment:
I think we are missing allowSignupCheck here altogether here. Its missing in main codebase also. Need to fix that immediately from our side. 

CC: @maidul98

Reply:
I can add it in this PR